### PR TITLE
Employee vertical slice JAVA

### PR DESCRIPTION
**Pull request by Conaill4**

### CODE REVIEW

**Review by ldevlin559 — APPROVED** [review]

File: src/main/java/org/example/controllers/EmployeeController.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+@Path("/api")
+public class EmployeeController {
+
+    EmployeeService employeeService;
+
+    public EmployeeController(EmployeeService employeeService) {
+        this.employeeService = employeeService;
+    }
+
+//    @POST
```
Comment (mlynch147) [inline src/main/java/org/example/controllers/EmployeeController.java:25]: Please remove commented out code

File: src/main/java/org/example/models/Employee.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    public Employee(int employeeID, String name, double salary, int bankAccountNumber
+    ,String nationalInsuranceNumber) {
+        EmployeeID = employeeID;
+        EmployeeName = name;
+        BankAccountNumber = bankAccountNumber;
+        Salary = salary;
+        NationalInsuranceNumber = nationalInsuranceNumber;
+    }
+
+    public Employee(int employeeID, String name, double salary, String bankAccountNumber, String nationalInsuranceNumber, int roleID) {
```
Comment (mlynch147) [inline src/main/java/org/example/models/Employee.java:24]: empty constructor

File: src/main/java/org/example/models/Employee.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    public void setBankAccountNumber(int bankAccountNumber) {
+        BankAccountNumber = bankAccountNumber;
+    }
+
+    public void setEmployeeName(String employeeName) {
+        EmployeeName = employeeName;
+    }
+
+    private Set<EmployeeRole> roles = new HashSet<>();
```
Comment (mlynch147) [inline src/main/java/org/example/models/Employee.java:64]: keep this up at the top with other properties

File: src/main/java/org/example/models/Employee.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    public Employee(){
```
Comment (mlynch147) [inline src/main/java/org/example/models/Employee.java:65]: keep the constructors together

File: src/main/java/org/example/models/Employee.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        this.roles = new HashSet<>();
+    }
+
+    public void addRole(EmployeeRole role){
```
Comment (mlynch147) [inline src/main/java/org/example/models/Employee.java:69]: not being used - remove for now

File: src/main/java/org/example/models/EmployeeRequest.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
+package org.example.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class EmployeeRequest {
+    ;
```
Comment (mlynch147) [inline src/main/java/org/example/models/EmployeeRequest.java:6]: rogue semi colon

File: src/main/java/org/example/services/EmployeeService.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+    EmployeeDao employeeDao;
+
+    public EmployeeService(EmployeeDao employeeDao) {
+        this.employeeDao = employeeDao;
+    }
+
+    public int createDeliveryEmployee(EmployeeRequest employeeRequest)
+            throws SQLException {
+        int Employeeid = employeeDao.createDeliveryEmployee(employeeRequest);
```
Comment (mlynch147) [inline src/main/java/org/example/services/EmployeeService.java:18]: lower case E please :)

File: src/main/java/org/example/services/EmployeeService.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+        if (Employeeid == -1) {
+            throw new RuntimeException("Error creating delivery employee");
```
Comment (mlynch147) [inline src/main/java/org/example/services/EmployeeService.java:21]: never throw a runtime exception - create a "CouldNotCreate" exception instead

File: src/main/java/org/example/services/EmployeeService.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+        }else{
+            int successid = employeeDao.insertRole(Employeeid, employeeRequest);
+
+            if (successid == -1) {
+                throw new RuntimeException("Error creating delivery employee code: 2");
```
Comment (mlynch147) [inline src/main/java/org/example/services/EmployeeService.java:26]: see above - create appropriate exception class

**Review by mlynch147 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**